iothub/device/src/Pipeline/TransportDelegatingHandler.cs

```diff
@@ -27,47 +27,27 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
 
             cancellationToken.ThrowIfCancellationRequested();
 
+            await _handlerLock.WaitAsync(cancellationToken).ConfigureAwait(false);
             try
             {
-                await _handlerLock.WaitAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
semaphore waits should happen outside the try/finally

Reply:
If we move it outside the try then the release on finally won't happen on exception. Is that what we want?

Reply:
That's not how it works. If we successfully get the lock, then we enter the try. If we fail to get the lock then you don't want to execute release, so we don't put it inside the try.